clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -756,10 +756,15 @@ protected boolean onJoinPrepare(Timer timer, int generation, String memberId) {
             joinPrepareTimer.update();
         }
 
+        final SortedSet<TopicPartition> partitionsToRevoke = getPartitionsToRevoke(protocol, generation, memberId);
```

Comment:
Do we need to recompute the partitions right before the revocation?

Reply:
Right, because if the consumer is assigned to a new partition, we will need to revoke them for eager protocol. (one case i can think of).

Reply:
I am not sure how we could be assigned new partitions before completing revocation. Is that possible? 

Another case from `getPartitionsToRevoke` below is when the subscription changes in the cooperative protocol. We use the current subscription to filter the partitions that can be revoked immediately before rejoining the group. But for partitions which are no longer subscribed, there is no need to send an offset commit and there is no danger of fetches returning. So I'm wondering if that case can be handled separately. In other words, can we delay computation of this set for the cooperative protocol until we are ready to call `onPartitionsRevoked`?

Reply:
i think it makes sense.  We probably can compute the partition during the revocation due to the nature of cooperative protocol.